From patchwork Tue May 12 11:02:56 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: =?utf-8?b?5b2t5rWpKFJpY2hhcmQp?=
 <richard.peng@oppo.com>
X-Patchwork-Id: 11542985
Return-Path: <SRS0=B8Dm=62=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 095C8912
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 May 2020 11:03:16 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id DF985206A3
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 12 May 2020 11:03:15 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (1024-bit key) header.d=oppoglobal.onmicrosoft.com
 header.i=@oppoglobal.onmicrosoft.com header.b="HTGPSVnR"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1729294AbgELLDE (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 12 May 2020 07:03:04 -0400
Received: from mail-eopbgr1300059.outbound.protection.outlook.com
 ([40.107.130.59]:9521
        "EHLO APC01-HK2-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S1727783AbgELLDD (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 12 May 2020 07:03:03 -0400
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=mlHoNdVWnv0Fa0r3A0utGCyROaMrURTIeuBJYNjvPNpXj30YbqpQj5JHNMIAsQEeOL7aDQ+yqRIAeGej3f7U08scFIbRuRNyZm17iYom2JWyNYg1QyVSEc8u0tFwmgidOFW6RXxsfN823ufYE0tABiIj4et2rq5JXCRZjGdUe/aTHZT58lo3a9Xk/B9qMFjdUwJPjz90lV+GeGYDmftWK9bw5/rm9FONo/Kl0XwlhbqIMrLOptisuNOR/+FEc40AAf+jAoGoIkOKMhX97lwD09k1SDpi4khXw45OHEJ/FKuychh3TwE+SjoXMaPcp+IDz8bv/ld005zEgvbLP/WvmA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=9+mNRqUwfmUoId8N83J4eXXZQmVldvgJiRRQpdxSnTs=;
 b=XOsdcOjYeSKthB9EC+5pKpyjw+UrynggpSzrwD3ViRPXHAfGYBseQs40KWDyh7aaDESqiYK5O7zQsbIbf55KWl5DI0y5qDMonX9PjZaugQ/LuoRbnmiR9uVYN88h34Giidw1PQKDGVwJ2z2xeA31q44wYE9rLPUdg2l4BgkIFjSuxTIz7uS+lA6kz6fX5iJJbJt12BD8amUap8dRWdLK9KwDdw3kKMdyXr4fple5zoyN4gqte2B6a/u1wFp2ZPBbwU6lb5uN6LiY6qbuLT9XKWiV1kA3Y4uQjOUuL0hI4F9pB7tHwy8CLYYr5YxDnWfDXRVlZj7lcdpj8p26dejqNw==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oppo.com; dmarc=pass action=none header.from=oppo.com;
 dkim=pass header.d=oppo.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oppoglobal.onmicrosoft.com; s=selector1-oppoglobal-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=9+mNRqUwfmUoId8N83J4eXXZQmVldvgJiRRQpdxSnTs=;
 b=HTGPSVnRK4ltqTDGsckdueir87UBT1WVSBkgl9Qj+VF1SqUZGksoZDrw5XYowFaO2FW0auz/t6yHzixR97D/HExb8voBZF6kghfrgpv3grAynzpoJORZEZuc901tIh208OMNX8JsruMPPhXSHcFfyI6eBbuasdSLeZeZK9eZI2k=
Received: from HKAPR02MB4291.apcprd02.prod.outlook.com (2603:1096:203:d3::12)
 by HKAPR02MB4308.apcprd02.prod.outlook.com (2603:1096:203:dd::8) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.2979.28; Tue, 12 May
 2020 11:02:58 +0000
Received: from HKAPR02MB4291.apcprd02.prod.outlook.com
 ([fe80::80ca:c698:988e:a999]) by HKAPR02MB4291.apcprd02.prod.outlook.com
 ([fe80::80ca:c698:988e:a999%5]) with mapi id 15.20.2979.033; Tue, 12 May 2020
 11:02:57 +0000
From: =?eucgb2312_cn?b?xe26xihSaWNoYXJkKQ==?= <richard.peng@oppo.com>
To: Paolo Bonzini <pbonzini@redhat.com>
CC: "linux-kernel@vger.kernel.org" <linux-kernel@vger.kernel.org>,
        "kvm@vger.kernel.org" <kvm@vger.kernel.org>
Subject: [PATCH v2] kvm/x86 : Remove redundant function implement
Thread-Topic: [PATCH v2] kvm/x86 : Remove redundant function implement
Thread-Index: AdYoTDpIz7AFtnDsRCiyAgzxuRMr2g==
Date: Tue, 12 May 2020 11:02:56 +0000
Message-ID: 
 <HKAPR02MB4291B0991BF57A5CCB1C7437E0BE0@HKAPR02MB4291.apcprd02.prod.outlook.com>
Accept-Language: zh-CN, en-US
Content-Language: zh-CN
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
authentication-results: redhat.com; dkim=none (message not signed)
 header.d=none;redhat.com; dmarc=none action=none header.from=oppo.com;
x-originating-ip: [58.255.79.102]
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 6b97a50f-9945-4101-81ed-08d7f6640703
x-ms-traffictypediagnostic: HKAPR02MB4308:
x-microsoft-antispam-prvs: 
 <HKAPR02MB4308CA3DE6683B45FA521EB4E0BE0@HKAPR02MB4308.apcprd02.prod.outlook.com>
x-ms-oob-tlc-oobclassifiers: OLM:119;
x-forefront-prvs: 0401647B7F
x-ms-exchange-senderadcheck: 1
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 s1SLYJ7Twdo29kTyMCq+iIh4GSg7xZpjk13pu6gAte1fK5ZQd/CcZ+NLJfIlUNH+/qUzaJNhIW9PrAi5zNDof/cSzmQ99/+b3sIXNO4rCZoeBzzXu27r81K8BH1Tm5uVUZm8lUdTeMkgMS1BVbBBYZ9SnLTHUDYG0LjcSExEOEeOykL/OQjokKYHirC9XdD6r1s2ZZZeIshmjNLVmSfLXnU8Fie1T/H9l4b5LznqmlCZFZ6C+VjI/Vh/EYZ/tfj05GgGoBC6MAPgiP8dTee0Jlr92H8odxB0mbSnzT/6xnYrMoZe8ygtMPf6GuvXDIZOD0fpxgYDW+z7pIUDtoM2ej+lJFIDmN7U9YvYPvoU9PIgCki0YQ1aE1algEmldd1B7h2xKm1SBVpGVIFzhrESKK8gGiw5+TtBuok7v/g/NCLqlwRfm5+OBTQH48H6T+x5JbmFu8Lkn68mXlpQdR10BKz/SfOvcY5o6lJwb5QjueLKQVdOOhQAE5KmdSw8yBJutZuvBGupbqRHsdBenphGnq7djODH8JefmCVC8eHjfezsgbgsb1QQyX5fRG9IyfEM
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:HKAPR02MB4291.apcprd02.prod.outlook.com;PTR:;CAT:NONE;SFTY:;SFS:(376002)(396003)(346002)(39860400002)(366004)(136003)(33430700001)(33440700001)(6506007)(55016002)(9686003)(52536014)(186003)(4326008)(508600001)(316002)(26005)(33656002)(7696005)(71200400001)(66946007)(86362001)(8676002)(85182001)(6916009)(64756008)(2906002)(66556008)(66476007)(8936002)(66446008)(54906003)(5660300002)(76116006)(11606004);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata: 
 mVZ3ZLT+4R/tgCIaJrWWZtg4NU7TzRIPqylR66VmoYv+i0Y1N0wXTlwt3wdnmTaqiJ33x8xsUkpi/meFgF7TMVOHRmSnld+gLWKmKZT2P14xesroJm2HJ03dyBRzMtaIFk0QR5iD8KY/u2daFXmxYXe72NUUVJs6ZyN4dwYTS6Nawi4UHplEdw+TOz/iiBZxoHjzNQDenAYQ68OshLjLaMpXHHlPxO6R99nNLbcrdElwCaFUQDPGK+wvlauC9t2XR7qVlXboA7rLOdtK6+lJRrGMblwHno2mhSN1ZCXhQhKgYZwUgVbKpCcx/kmEEYJHIKCpo5kN+2hqi+Vp+zECtw7qUH5t3zeP2IBrqw5+Rc/JAi7LCGV+9QyPar4Aq6hmCooUQBZka1zQ+hBGN9fYR/ftJI30HO5Ilp3IQI+tAUSxTElW/jcWCVoXsJze7lfQoF+Jlr8PAHyOC2VqpGUN9kNi+pvnXLKDeysT9y6HVeY=
x-ms-exchange-transport-forked: True
MIME-Version: 1.0
X-OriginatorOrg: oppo.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 6b97a50f-9945-4101-81ed-08d7f6640703
X-MS-Exchange-CrossTenant-originalarrivaltime: 12 May 2020 11:02:56.7084
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: f1905eb1-c353-41c5-9516-62b4a54b5ee6
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 4mlCwaxHDitAxkgUAFMsGxzUqrUvLtaM52pR6k9Bmt7wAKU44StF2TjtbaVj7x+XPoVf4/RJBwebUqXUKam7xA==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: HKAPR02MB4308
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

pic_in_kernel(),ioapic_in_kernel() and irqchip_kernel() have the
same implementation.

Signed-off-by: Peng Hao <richard.peng@oppo.com>
---
 arch/x86/kvm/ioapic.h |  8 ++------
 arch/x86/kvm/irq.h    | 13 ++++---------
 2 files changed, 6 insertions(+), 15 deletions(-)

--
2.7.4

diff --git a/arch/x86/kvm/ioapic.h b/arch/x86/kvm/ioapic.h
index 2fb2e3c..7a3c53b 100644
--- a/arch/x86/kvm/ioapic.h
+++ b/arch/x86/kvm/ioapic.h
@@ -5,7 +5,7 @@
 #include <linux/kvm_host.h>

 #include <kvm/iodev.h>
-
+#include "irq.h"
 struct kvm;
 struct kvm_vcpu;

@@ -108,11 +108,7 @@ do {\

 static inline int ioapic_in_kernel(struct kvm *kvm)
 {
-int mode = kvm->arch.irqchip_mode;
-
-/* Matches smp_wmb() when setting irqchip_mode */
-smp_rmb();
-return mode == KVM_IRQCHIP_KERNEL;
+return irqchip_kernel(kvm);
 }

 void kvm_rtc_eoi_tracking_restore_one(struct kvm_vcpu *vcpu);
diff --git a/arch/x86/kvm/irq.h b/arch/x86/kvm/irq.h
index f173ab6..c0ad731 100644
--- a/arch/x86/kvm/irq.h
+++ b/arch/x86/kvm/irq.h
@@ -66,15 +66,6 @@ void kvm_pic_destroy(struct kvm *kvm);
 int kvm_pic_read_irq(struct kvm *kvm);
 void kvm_pic_update_irq(struct kvm_pic *s);

-static inline int pic_in_kernel(struct kvm *kvm)
-{
-int mode = kvm->arch.irqchip_mode;
-
-/* Matches smp_wmb() when setting irqchip_mode */
-smp_rmb();
-return mode == KVM_IRQCHIP_KERNEL;
-}
-
 static inline int irqchip_split(struct kvm *kvm)
 {
 int mode = kvm->arch.irqchip_mode;
@@ -93,6 +84,10 @@ static inline int irqchip_kernel(struct kvm *kvm)
 return mode == KVM_IRQCHIP_KERNEL;
 }

+static inline int pic_in_kernel(struct kvm *kvm)
+{
+return irqchip_kernel(kvm);
+}
 static inline int irqchip_in_kernel(struct kvm *kvm)
 {
 int mode = kvm->arch.irqchip_mode;
